public/js/navigator-js/integration/StateViewMap.js

```diff
@@ -43,33 +45,90 @@ this.navigatorjs.integration = this.navigatorjs.integration || {};
 				state = recipeStates[j];
 
 				if (requestedState.contains(state)) {
-					viewInstance = recipe.getViewInstance();
 
-					if (viewInstance.navigatorBehaviors instanceof Array) {
-						_addViewElementToDOM(recipe);
+					_addViewElementToDOM(recipe);
+					var viewInstance = recipe.getViewInstance();
+					if (Array.isArray(viewInstance.navigatorBehaviors)) {
 						_navigator.add(viewInstance, state);
 					}
+
+				} else {
+					if (recipe.isMounted()) {
+						_removeViewElementFromDOM(recipe);
+					}
 				}
 			}
 		}
 	}
 
+	function _removeViewElementFromDOM(recipe) {
```

Comment:
> the .remove() method takes elements out of the DOM. Use .remove() when you want to remove the element itself, as well as everything inside it. In addition to the elements themselves, all bound events and jQuery data associated with the elements are removed. To remove the elements without removing data and events, use .detach() instead.

Ran https://jsperf.com/removing-an-element-from-the-screen/3 several times and got an average of 18% slower with adding / removing DOM elements compare to display none/block - Its pretty significant.